### ISSUE TITLE
improve the reject error message

### ISSUE THREAD

**Issue by vincenzopalazzo** [daywalker90/clnrod#1]
Having an error during the opening channel is always bad, but this time if what we wanted.

However, the error message send by cln to the node when rejecting the  is really strange

```
➜  ~ lightning-cli fundchannel pubkey all
{
   "code": -1,
   "message": "They sent ERROR channel pubkey: You gave bad parameters: ",
   "data": {
      "id": "pubkey",
      "method": "fundchannel_start"
   }
}
```

We should return a better error message instead of `You gave bad parameters: `, in this way the user know that we are applying some sort of filtering

**Comment by daywalker90** [comment]
Yes, but i can only control what comes after `They sent ERROR channel <pubkey>: You gave bad parameters:` e.g. if you set the deny message to `Sorry, i don't want a channel from you!` it will come out as `They sent ERROR channel <pubkey>: You gave bad parameters: Sorry, i don't want a channel from you!`. The first part is from cln not me.

**Comment by daywalker90** [comment]
`clnrod-denymessage` option defaults to an empty string that's why you don't see anything there.

**Comment by vincenzopalazzo** [comment]
>Yes, but i can only control what comes after They sent ERROR channel <pubkey>: You gave bad parameters: e.g. if you set the deny message to Sorry, i don't want a channel from you! it will come out as They sent ERROR channel <pubkey>: You gave bad parameters: Sorry, i don't want a channel from you!. The first part is from cln not me.

Ok I will open a patch to CLN so.

>clnrod-denymessage option defaults to an empty string that's why you don't see anything there.

it is possible to set the default to another value?

**Comment by daywalker90** [comment]
> Ok I will open a patch to CLN so.

Great!

> it is possible to set the default to another value?

Do you mean this: `lightning-cli setconfig clnrod-denymessage "Channel rejected!"`. Or the actual default for every fresh installation? Both can be done. But what to choose for everyone 🤔

**Comment by vincenzopalazzo** [comment]
>Do you mean this: lightning-cli setconfig clnrod-denymessage "Channel rejected!". Or the actual default for every fresh installation? Both can be done. But what to choose for everyone 🤔

Any message that informs other parties that this is a custom rejection, not a protocol mismatch, would be good. If the user want a custom one will set their own reason

I would like to avoid situations where people start to claim that Lightning is broken if we don't return a custom message. For example, I am currently debugging an LND bug where I cannot open a private channel from CLN. It would be helpful to easily distinguish between different causes for these two messages.

**Comment by daywalker90** [comment]
Fixed in v0.2.1
